### PR TITLE
feat(batch): Batch processing wrapper function

### PR DESCRIPTION
**Pull request by erikayao93**
## Description of your changes
- Renamed `itemIdentifier` and `batchResponse` types to match Python naming
- Renamed testing factories to be more accurate to their function
- Added `processPartialResponse()` function that acts as a function wrapper for batch processing
- Added unit tests for `processPartialResponse()` function
- Added some `TODO` comments for future async/sync function work

### Related issues, RFCs

**Issue number: #1602** 

## Checklist

- [x] [My changes meet the tenets criteria](https://docs.powertools.aws.dev/lambda-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [x] My changes generate *no new warnings*
- [x] I have *added tests* that prove my change is effective and works
- [x] The PR title follows the [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/batch/src/processPartialResponse.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+} from '.';
+
+const processPartialResponse = async (
+  event: { Records: BaseRecord[] },
+  recordHandler: CallableFunction,
+  processor: BasePartialBatchProcessor
+): Promise<PartialItemFailureResponse> => {
+  const records: BaseRecord[] = event['Records'];
+
+  if (!records) {
```
Comment (am29d) [inline packages/batch/src/processPartialResponse.ts:15]: I think we can call `!event.Records` here without the const declaration.

**Review by am29d — COMMENTED** [review]
Hey @erokayao93, looks great! I left just one comment, otherwise we can merge it.

File: packages/batch/tests/unit/processPartialResponse.test.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    });
+
+    test('Process partial response through handler for SQS records with incorrect event type', async () => {
+      // Prepare
+      const processor = new BatchProcessor(EventType.SQS);
+      const event = dummyEvent;
+      const eventTypes: string = Object.values(EventType).toString();
+
+      const handler = async (
+        event: any,
```
Comment (dreamorosi) [inline packages/batch/tests/unit/processPartialResponse.test.ts:176]:
```suggestion
        event: SQSEvent,
```

File: packages/batch/tests/unit/processPartialResponse.test.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        _context: Context
+      ): Promise<PartialItemFailureResponse> => {
+        return await processPartialResponse(event, sqsRecordHandler, processor);
+      };
+
+      // Act & Assess
+      await expect(handler(event, context)).rejects.toThrowError(
```
Comment (dreamorosi) [inline packages/batch/tests/unit/processPartialResponse.test.ts:183]:
```suggestion
      await expect(handler(event as unknown as SQSEvent, context)).rejects.toThrowError(
```

**Review by dreamorosi — COMMENTED** [review]
I have left two suggestions, if you accept them you should be able to remove the `eslint-disable` from the file.

**Review by dreamorosi — APPROVED** [review]
Thanks for the PR and for the quick turnaround!

**Pull request merged**